connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -908,7 +908,7 @@ public Void call() throws Exception {
                 } else if (isLeader()) {
                     callback.onCompletion(new NotAssignedException("Cannot restart connector since it is not assigned to this member", member.ownerUrl(connName)), null);
                 } else {
-                    callback.onCompletion(new NotLeaderException("Cannot restart connector since it is not assigned to this member", leaderUrl()), null);
+                    callback.onCompletion(new NotLeaderException("Cannot restart connector since it is not leader", leaderUrl()), null);
```

Comment:
Same concern here about this change being less clear, as mentioned in my other comment.